src/clr-angular/forms/datepicker/date-input.spec.ts

```diff
@@ -0,0 +1,288 @@
+                   fixture.detectChanges();
+
+                   expect(ioService.date).toBeNull();
+               }));
+
+            it("updates the input element value when the date is updated", () => {
+                const ioService: DateIOService = dateContainerDebugElement.injector.get(DateIOService);
+
+                expect(fixture.componentInstance.dateValue).toBeUndefined();
+
+                const date = new Date(2015, 1, 1);
+                ioService.updateDate(date);
+
+                fixture.detectChanges();
+
+                expect(fixture.componentInstance.dateValue).toBe("02/01/2015");
+            });
+        });
+
+        describe("Datepicker with clrDate", () => {
+            let fixture: ComponentFixture<TestComponentWithClrDate>;
+            let compiled: any;
+            let dateContainerDebugElement: DebugElement;
+            let dateInputDebugElement: DebugElement;
+
+            beforeEach(function() {
+                TestBed.configureTestingModule(
+                    {imports: [FormsModule, ClrFormsModule], declarations: [TestComponentWithClrDate]});
+
+                fixture = TestBed.createComponent(TestComponentWithClrDate);
+                fixture.detectChanges();
+                compiled = fixture.nativeElement;
+                dateContainerDebugElement = fixture.debugElement.query(By.directive(ClrDateContainer));
+                dateInputDebugElement = fixture.debugElement.query(By.directive(ClrDateInput));
+            });
+
+            it("supports a clrDate Input", () => {
+                const ioService: DateIOService = dateContainerDebugElement.injector.get(DateIOService);
+                expect(ioService.date).toBeUndefined();
+
+                const date: Date = new Date();
+
+                fixture.componentInstance.date = date;
+
+                fixture.detectChanges();
+
+                expect(ioService.date.getFullYear()).toBe(date.getFullYear());
+                expect(ioService.date.getMonth()).toBe(date.getMonth());
+                expect(ioService.date.getDate()).toBe(date.getDate());
+            });
+
+            it("emits the output date correctly", () => {
+                const ioService: DateIOService = dateContainerDebugElement.injector.get(DateIOService);
+                expect(ioService.date).toBeUndefined();
+                expect(fixture.componentInstance.date).toBeUndefined();
+
+                const date: Date = new Date();
+                ioService.updateDate(date);
+                fixture.detectChanges();
+
+                expect(fixture.componentInstance.date.getFullYear()).toBe(date.getFullYear());
+                expect(fixture.componentInstance.date.getMonth()).toBe(date.getMonth());
+                expect(fixture.componentInstance.date.getDate()).toBe(date.getDate());
+            });
+
+            it("emits the date when the user changes the input", () => {
```

Comment:
When I type a date the service gets updated with the date - view test

Reply:
done